docs/creating-a-runtime-check.md

```diff
@@ -0,0 +1,162 @@
+
+```php
+use WordPress\Plugin_Check\Checker\Check_Result;
+use WordPress\Plugin_Check\Checker\Checks\Abstract_Runtime_Check;
+use WordPress\Plugin_Check\Traits\Stable_Check;
+
+class My_Custom_Check extends Abstract_Runtime_Check {
+
+  use Stable_Check;
+  
+  public function get_categories() {
+    // Return an array of check categories.
+    // See the `WordPress\Plugin_Check\Checker\Check_Categories` class for available categories.
+  }
+
+  public function prepare() {
+    // Handle the Checks preparations and return a cleanup function.
+  }
+
+  public function run( Check_Result $result ) {
+    // Handle running the check and adding warnings or errors to the result.
+  }
+}
+```
+
+## Preparations
+
+Preparations in the Plugin Checker are used to set up the environment before running checks, involving logic like adding actions and filters, and creating necessary test content. They utilize the `Preparation` interface, requiring the implementation of a `prepare()` method, and return cleanup functions to revert changes made during preparation.
+Three different approaches can be used for creating preparations based on the circumstances:
+
+* Check preparations
+* Shared preparations
+* Global preparations
+
+### Check preparations
+
+Check preparations are created using the `prepare()` method within a check class.
+
+This type of preparation is useful for carrying out the logic to prepare the environment that is specific to the check that implements it.
+
+Below is an example of a preparation that creates a test post and returns a cleanup function that deletes that post after the check is run.
+
+```php
+use WordPress\Plugin_Check\Checker\Checks\Abstract_Runtime_Check;
+
+class My_Custom_Check extends Abstract_Runtime_Check {
+  /**
+   * Check Preparation.
+   */
+  public function prepare() {
+    // Create a test post ahead of running the check.
+    $post_id = wp_insert_post(
+      array(
+        'post_title'   => 'Test post',
+        'post_content' => 'This is a test post.',
+        'post_type'    => 'post',
+      )
+    );
+
+    // Return the clean up function.
+    return function() use ( $post_id ) {
+      // Remove the test post created.
+      wp_delete_post( $post_id );
+    }
+  }
+}
+```
+
+### Shared Preparations
+
+Shared preparations are individual classes created to handle specific preparation logic in a single place.
+
+The [`Demo_Posts_Creation_Preparation`](https://github.com/10up/plugin-check/blob/trunk/includes/Checker/Preparations/Demo_Posts_Creation_Preparation.php) in the Plugin Checker is an example of a preparation class.
+
+Preparation classes are useful for implementing the same preparation logic across multiple checks using the concept of shared preparations.
+
+Shared preparations are used to prevent running the same preparations multiple times.
+
+Before running checks against a plugin all shared preparations are collected and processed to remove any shared preparations that are the same. 
```

Comment:
```suggestion
Before running checks against a plugin all shared preparations are collected and processed to avoid duplicate execution of any shared preparations that are the same. 
```